addons/payment/static/src/js/payment_form.js

```diff
@@ -10,6 +10,7 @@ var _t = core._t;
 publicWidget.registry.PaymentForm = publicWidget.Widget.extend({
     selector: '.o_payment_form',
     events: {
+        'submit': 'async payEvent',
```

Comment:
what happens what it's a payment management form? need to trigger the other handler instead...

Reply:
add the `mode` variable in the form `data` attributes to have it in the widget's options

trigger both handlers

have early returns if it's not the mode they're supposed to handle

i think this should work